api/edge_api/identities/utils.py

```diff
@@ -0,0 +1,40 @@
+import typing
+
+from flag_engine.features.models import FeatureStateModel
+
+if typing.TYPE_CHECKING:
+    from edge_api.identities.types import ChangeType, FeatureStateChangeDetails
+
+
+def generate_change_dict(
+    change_type: "ChangeType",
+    identity_id: int | str | None,
+    new: FeatureStateModel | None = None,
+    old: FeatureStateModel | None = None,
+) -> "FeatureStateChangeDetails":
+    if not (new or old):
+        raise ValueError("Must provide one of 'new' or 'old'")
+
+    change_dict = {"change_type": change_type}
+    if new:
+        change_dict["new"] = _get_overridden_feature_state_dict(
+            identity_id=identity_id,
+            feature_state=new,
+        )
+    if old:
+        change_dict["old"] = _get_overridden_feature_state_dict(
+            identity_id=identity_id,
+            feature_state=old,
+        )
+
+    return change_dict
+
+
+def _get_overridden_feature_state_dict(
+    identity_id: int | str | None,
+    feature_state: FeatureStateModel,
+) -> dict[str, typing.Any]:
+    return {
+        **feature_state.dict(),
+        "feature_state_value": feature_state.get_value(identity_id),
```

Comment:
Just making a note here as well that this is different to the previous behaviour - this used to be `"value"` I think?

Reply:
From what I can tell, this data is used in the `generate_audit_log_records` task (as well as the new task you created) but the value is not actually used so I think this is fine.

Reply:
Yes, `"value"` was not used. I made sure to modify the `generate_audit_logs` tests accordingly.